src/flowchem/devices/knauer/knauer_autosampler.py

```diff
@@ -174,288 +127,31 @@ def _send_and_receive(self, message: str):
                 f"No Connection possible to device with ip_address {self.ip_address}"
             )
 
-class Vial:
-        
-    # TODO get the rounding issue of ureg right
-    def __init__(self, substance, solvent:str or None, concentration: str, contained_volume:str, remaining_volume:str):
-        self._remaining_volume = flowchem_ureg(remaining_volume)
-        self._contained_volume = flowchem_ureg(contained_volume)
-        self.substance = set_vial_content(substance) # todo get this canonical
-        if not check_special_vial(substance):
-            self.solvent = solvent
-            self.concentration = flowchem_ureg(concentration)
-        else:
-            self.solvent = None
-            self.concentration = None
-
-    def extract_from_vial(self, volume:str):
-        if type(volume) is str:
-            volume = flowchem_ureg(volume)
-        self._contained_volume -= volume
-    
-    @property
-    def available_volume(self)-> flowchem_ureg.Quantity:
-        return self._contained_volume-self._remaining_volume
-
-
-class TrayPosition:
-    def __init__(self, side, row, column):
-        self.side = side
-        self.row = row
-        self.column = column
-        self.valid_position()
-
-    def valid_position(self):
-        from numpy import int64
-        assert self.side.upper() == SelectPlatePosition.LEFT_PLATE.name or self.side.upper() == SelectPlatePosition.RIGHT_PLATE.name
-        assert type(self.row) in [int64, int]
-        assert type(self.column) == str and len(self.column) == 1
 
 
-    # basically, only acts a s container internally and to make substance access easy
-
-# TODO create a Tray class
-class Tray:
```

Comment:
Tray and Vial are important for xyz gantry imho - there needs to be a grid and the grid positions have contents

Reply:
Or you would change the xyz component and add these two classes? Or maybe a new child component "Autosampler" that uses Vial() and Tray(). (Because a 3D printer wouldn't need these classes for example)